tox.ini

```diff
@@ -56,42 +66,59 @@ exclude = .tox,wiki,.cache,.d3a,.hypothesis,.pytest_cache,vagrant,requirements,v
 basepython = python3.8
 passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH SOLC_BINARY
 deps =
-	pip-tools
-	coverage
+    pip-tools
+    coverage
 commands =
     pip-sync requirements/tests.txt
     pip install -e .
-    coverage run -m py.test --random-order {posargs:tests}
+    coverage run -m pytest --random-order {posargs:tests}
     coverage combine
     coverage report
 
 [testenv:ci]
 basepython = python3.8
-passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH SOLC_BINARY BRANCH
+passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH SOLC_BINARY GSY_FRAMEWORK_BRANCH INTEGRATION_TESTS_REPO
 deps =
-	pip-tools
-	coverage
-whitelist_externals = bash
-
-commands =
-	python -c "import fcntl; fcntl.fcntl(1, fcntl.F_SETFL, 0)"
+    pip-tools
+    coverage
+allowlist_externals =
+    git
+    /bin/cp
+    /bin/rm
+    bash
+commands_pre =
+    python -c "import fcntl; fcntl.fcntl(1, fcntl.F_SETFL, 0)"
     pip-sync requirements/tests.txt requirements/pandapower.txt
     pip install -e .
-	pip uninstall -y gsy-framework
-	pip install git+https://github.com/gridsingularity/gsy-framework@{env:BRANCH:master}
+    pip uninstall -y gsy-framework
+    pip install git+https://github.com/gridsingularity/gsy-framework@{env:GSY_FRAMEWORK_BRANCH:master}
+commands =
     flake8
-    coverage run -m py.test --random-order {posargs:tests}
+    coverage run -m pytest --random-order {posargs:tests}
     coverage combine
     coverage xml
     coverage report
+    git clone {env:INTEGRATION_TESTS_REPO:git@github.com:gridsingularity/gsy-backend-integration-tests.git} {envtmpdir}/gsy-backend-integration-tests
+    cp -r {envtmpdir}/gsy-backend-integration-tests/gsy_e_tests/integration_tests/ ./integration_tests/
```

Comment:
I tried it out locally and works smoothly.
However, I'd like to mention that the reason behind removing the copied folder (or in this case, the softlink) on `commands_post` was to leave the working folder (cloned repo) clean (just as it was before the execution) to avoid any confusion or accidental commit of these files where working locally.
So, IMO, `commands_post` should continue to be there.
I'm open to know anyone's opinion on this matter. What do you think?

Reply:
I agree, does not hurt to leave the post step here in order to end up with a clean repository. Thanks for dealing with it!